### ISSUE TITLE
Make it easier to clear localStorage on web property

### ISSUE THREAD

**Issue by caitmuenster** [mozilla/FirefoxColor#586]
Reported on behalf of Madonna, one of our theme power users. 

Madonna has been using Firefox Color to preview her themes before she publishes them to AMO. On November 12, she reported that she was not able to make new changes or create new themes using Firefox Color. I think she ended up refreshing her Firefox. 

Today, she was able to 10 images before she was no longer able to make changes or add new themes. 

Madonna, could you try clearing your cache one more time, and if that doesn't work, can you post a screenshot of any error messages you see?

**Comment by MaDonnasPersonas** [comment]
There is no error message.

**Comment by lmorchard** [comment]
TL;DR: Assuming this is a problem with storage space, the immediate workaround for this issue is probably to 1) remove the Color add-on and then 2) [clear storage for color.firefox.com](https://support.mozilla.org/en-US/kb/storage). This should start things from a fresh slate.

In the meantime, we can look at changes to display available space remaining and to more gracefully handle the error. Unfortunately, I don't think we can *expand* the available storage for images at present.

> There is no error message.

There might be an error message in the Web Console - [and here's how to see those errors](https://developer.mozilla.org/en-US/docs/Tools/Web_Console/Opening_the_Web_Console). It's probably about localStorage quota limits, but I'm not sure. Confirmation would be helpful. Though, I will try loading my browser with large images to see if I can reproduce the issue on my side.

> does the Firefox Color extension have unlimited data for background images?

No, we don't ask for the "unlimitedStorage" permission, [so the extension has a limit of 5MB](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local). But, even if we did have unlimited storage in the extension, we need also localStorage for the web-based theme editor. Unfortunately, [web localStorage has a limit of 10MB in Firefox](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API). So, we're probably stuck with that as an upper limit for now.

> I'm not sure how to clear it from the extension itself.

I don't think there is a way to clear the storage for an extension, short of removing the extension altogether and re-installing it.

**Comment by MaDonnasPersonas** [comment]
Thanks...

**Comment by caitmuenster** [comment]
Thanks, @lmorchard! We're going to give uninstalling/reinstalling a shot. Appreciate your willingness to help troubleshoot!

**Comment by johngruen** [comment]
i can try to repro to get an idea about what we can do...

@MaDonnasPersonas we are going to be adding an export feature meaning you can make themes that can be submitted to AMO rather than simply saving them locally. But at the same time we can improve error handling here to make your life easier!

**Comment by MaDonnasPersonas** [comment]
Thanks so much for the update. If I use CCleaner when FC won't let me add
any Custom backgrounds, then I can get it to work again after I close it
and open it up.

**Comment by caitmuenster** [comment]
It looks like we figured this out! Uninstalling the extension appears to not clear out local storage. 

We were able to solve this by going to color.firefox.com, opening the storage inspector, deleting all from Local Storage, uninstalling the extension, and then returning to color.firefox.com to refresh the extension. 

It would be great if we could make that process easier so folks don't need to go to the storage inspector to clear local storage.

**Comment by MaDonnasPersonas** [comment]
Yes, but it is real easy to do once you know how! We will just hope this
continues to work. Will keep in touch. I do so appreciate your help.
